drivers/include/lm75.h

```diff
@@ -0,0 +1,240 @@
+ * @return LM75_ERROR_I2C, on I2C related error
+ * @return LM75_ERROR, on temperature setting related error
+ */
+int lm75_set_temp_limits(lm75_t *dev, int temp_hyst, int temp_os, gpio_cb_t cb, void *arg);
+
+/**
+ * @brief Overshutdown temperature value of LM75 sensor
+ *
+ * Reads the sensor OS temperature value from TOS_REG in ºC.
+ *
+ * @param[in] dev                 device structure
+ * @param[out] temperature        buffer where OS temperature value will be written
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ */
+int lm75_get_os_temp(lm75_t *dev, int *temperature);
+
+/**
+ * @brief Hysteresis temperature value of LM75 sensor
+ *
+ * Reads the sensor hysteresis temperature value from THYST_REG in ºC.
+ *
+ * @param[in] dev                 device structure
+ * @param[out] temperature        buffer where HYST temperature value will be written
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ */
+int lm75_get_hyst_temp(lm75_t *dev, int *temperature);
+
+/**
+ * @brief Read the current state of the OS pin to see if it's active.
+ *
+ * Read the configuration register to see the OS pin's polarity and
+ * then reads its state. Then outputs if the pin is active and whether
+ * it's in the low and active or high and active.
+ *
+ * @param[in] dev      device structure
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR, on pin reading error
+ * @return LM75_I2C_ERROR, on I2C related error
+ */
+int lm75_get_os_pin(lm75_t *dev);
+
+/**
+ * @brief Activate the LM75 sensor shutdown mode
+ *
+ * @param[in] dev      device structure to set into shutdown mode
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR, on mode switching related error
+ */
+int lm75_poweroff(lm75_t *dev);
+
+/**
+ * @brief Deactivate the LM75 sensor shutdown mode
+ *
+ * @param[in] dev      device structure to wake up from shutdown mode
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR, on mode switching related error
+ */
+int lm75_poweron(lm75_t *dev);
+
+/**
+ * @brief Activates one shot conversion mode
+ *
+ * Wakes from shutdown mode, does a single temperature conversion
+ * and writes in into the temperature register and then goes back into shutdown
+ *
+ * @param[in] dev       device structure
+ *
+ * @return LM75_ERROR if unsuccessful
+ * @return LM75_SUCCESS if successful
+ */
+int tmp1075_one_shot(lm75_t *dev);
+
+/**
+ * @brief Activates low power mode operation
+ *
+ * This function makes the device measure temperatures in a strictly discrete way
+ * at a user definable rate, as opposed to performing continuous measurements
+ * at the device's conversion rate.
+ * It allows the device to stay in shutdown mode for the most part, therefore consuming less power.
+ * In the tmp1075 sensor this is done automatically using the one shot mode feature.
+ * In the LM75A sensor this is done manually by switching the device to and from shutdown mode.
```

Comment:
I still don't get why this function is needed.
Why can't we have a `lm75_read_oneshot()` that does

 - power on the device
 - wait for a single conversion / enable one-short mode
 - take a measurement
 - power off the device

Reply:
I decide to split the tmp1075's one shot mode away from the low power mode just to clarify that for the tmp1075 it's a native feature of the device and in the lm75a and other's its just emulated via the driver.